### DISCUSSION TITLE
Can hosts be created dynamically?

### DISCUSSION THREAD

**Discussion by russellfeeed** [deployphp/deployer#2798] [General]
My hosting provider has an APi which allows me to pull a listy of all hosts and applications,
Can I call host() iteratively to dynamicall create all the hosts?

I currently have something like this, which works if I call it from a task, but will not work called outside of a task
```
public function createHostsForAllServersAndApps() {
        $serversAndApps=$this->helpers->getAllServersAndApps();
        foreach ($serversAndApps->servers as $server) {
            //print_r($server);
            //die(print_r($server->apps,true));
            foreach ($server->apps as $app) {
                // create a deployer host for each app
                writeln("Creating Deployer Host for {$app->app_fqdn} / {$server->label} / {$app->sys_user} / {$server->master_user} ");
                host($app->app_fqdn)
                    ->set('hostname', $app->id) 
                    ->set('release_path', $app->sys_user) // get this from https://platform.cloudways.com/apps/XXXX/access_detail
                    ->set('remote_user', $server->master_user);
               

            } // foreach app
        } // foreach server
    } // createHostsForAllServersAndApps
```

**Comment by antonmedv** [comment]
Yes, just configure it outside of task.

**Comment by russellfeeed** [comment]
Thanks. It gives a pimple error about 'output'. Could this be due to the writeln() call do you think?

**Reply by antonmedv** [reply]
Yes. It works only inside tasks. Use echo.

**Comment by russellfeeed** [comment]
So finally got this working by replacing `runLocally('curl....')` with a `shell_exec('curl....')`. runLocally obviously only works in a task

**Comment by russellfeeed** [comment]
See https://github.com/deployphp/deployer/discussions/2635
